ark-activemq/templates/secrets.yaml

```diff
@@ -0,0 +1,531 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "ark-activemq.fullname" . }}
+  labels:
+    {{- include "ark-activemq.labels" . | nindent 4 }}
+type: Opaque
+stringData:
+  activemq.xml: |
+    <!--
+        Licensed to the Apache Software Foundation (ASF) under one or more
+        contributor license agreements.  See the NOTICE file distributed with
+        this work for additional information regarding copyright ownership.
+        The ASF licenses this file to You under the Apache License, Version 2.0
+        (the "License"); you may not use this file except in compliance with
+        the License.  You may obtain a copy of the License at
+    
+        http://www.apache.org/licenses/LICENSE-2.0
+    
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+    -->
+    <!-- START SNIPPET: example -->
```

Comment:
@naveen-armedia We discussed this in meetings. They have to be secrets if they contain secrets.

Reply:
@fabricetriboix  as far as i know activemq.xml and other things mentioned as secrets are config files. And as I remember we discussed using config files in  configMap object and  all the secrets like HashKeys and plain username and passwords goes in as secret object.

Reply:
@naveen-armedia `activemq.xml` contains secrets such as passwords. Unfortunately, there is no way to have part of the file as a ConfigMap and part of the file as Secret. Ditto for most of the other configuration files. I could potentially separate the configuration files that do not contain secrets into a ConfigMap, but I think it's easier to put all the config files in one Kubernetes object. Additionally, having files being mounted in the same directory from different Secrets/ConfigMaps is not that trivial and just couldn't be done in older versions of Kubernetes.

Reply:
@fabricetriboix as I see we have only username and passwords defined in `credentials.properties` and `jetty-realm.properties` files only. And in `activemq.xml`  we are just referring to the path where credentials.properties file is located. Same with other files `groups.properties` , `jetty.xml`, `log4j.properties`, `login.config`, `users.properties`.

Reply:
@naveen-armedia Yes, my bad. I meant some other files. Are you suggesting I put `activemq.xml` and other config files that don't contain secrets in a ConfigMap and to keep the other config files that do contain secrets as Secrets? If yes, could you please let me know how to mount both a ConfigMap and a Secret onto the same directory inside the container? Thanks

Reply:
@fabricetriboix Yes `groups.properties` , `jetty.xml`, `log4j.properties`, `login.config`, `users.properties`  files can be kept in the configMap objects. We can VolumeMount **each config item**  to the same directory with `subPath` like 

```yaml
          volumeMounts:
            - name: application-config
              mountPath: "/app/conf/activemq.xml"
              subPath: activemq.yml
              readOnly: true
```
I think it's not a good way to have whole configuration file `credentials.properties` as an item in secret object. Also, We could try adding  username and password as secret separately like
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
stringData:
    USERNAME: "Armedia-User"
    PASSWORD: "1234"
```
And use the credentials as env variables for the POD like below. Although still secrets are encoded with base64, we can figured out latter using sealed secrets with some Vaults as discussed earlier.
```yaml
apiVersion: apps/v1
kind: Deployment
metadata: 
  name: deployment
spec: 
  replicas: 1
  template: 
    metadata: 
      labels: 
        app: gateway
    spec: 
      containers: 
        - 
          env: 
            - 
              name: WSO2_CLOUD_ORG_KEY
              valueFrom:
                secretKeyRef:
                  name: mysecret
                  key: USERNAME
```
In the config file we can use this environment variable like below. 
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ include "ark-cloudconfig.fullname" . }}
  annotations:
    timestamp: "{{ date "20060102150405" .Release.Time }}"
data:
   credentials.properties: |-
    ## ---------------------------------------------------------------------------
    ## Licensed to the Apache Software Foundation (ASF) under one or more
    ## contributor license agreements.  See the NOTICE file distributed with
    ## this work for additional information regarding copyright ownership.
    ## The ASF licenses this file to You under the Apache License, Version 2.0
    ## (the "License"); you may not use this file except in compliance with
    ## the License.  You may obtain a copy of the License at
    ## 
    ## http://www.apache.org/licenses/LICENSE-2.0
    ## 
    ## Unless required by applicable law or agreed to in writing, software
    ## distributed under the License is distributed on an "AS IS" BASIS,
    ## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    ## See the License for the specific language governing permissions and
    ## limitations under the License.
    ## ---------------------------------------------------------------------------
    
    # Defines credentials that will be used by components (like web console) to access the broker
    
    activemq.username=${mysecret}
    activemq.password=${mysecret}
    guest.password=${mysecret}
```

Reply:
@naveen-armedia I was not aware we could use environment variables in ActiveMQ configuration files. Could you please point me to the ActiveMQ documentation that explain this? Thanks

Reply:
@fabricetriboix I don't know much experience on activemq. But we have used the way of environment variable  for java application and nodejs application  in the past. Also we made it like same for cloudconfig too. Please try and let me know if you are having issues using environment variable in activemq configuration files.

Reply:
@JonHolman Are you happy for me to spend some time trying out what Naveen suggested?

Reply:
@fabricetriboix  yes, please do.

Reply:
@naveen-armedia I haven't been able to use environment variables in config files, they are not expanded. Additionally, it is not possible by default to mount two or more files into the same directory. I believe recent versions of Kubernetes permit some kind of merging, but I don't think the added complexity is worth it.

Reply:
@fabricetriboix in which version of kubernetes are you trying..? My only concern is when these are supposed to be configurations we should not be using  these in secrets objects. I don't think implementation is much complex. I will raise an issue with improvement so that it could be taken up in next time. Lets just focus on how we can go with deploy script .Thanks!!

Reply:
@naveen-armedia Whether environment variables are expanded inside configuration files is a question of whether the Java class that is reading those config files preforms the expansion or not. I fail to understand how this has anything to do with Kubernetes...
Or perhaps you mentioned Kubernetes in relation to mounting multiple files into the same directory? This is actually something that Kubernetes already supports: https://kubernetes.io/docs/concepts/storage/volumes/#projected
I just said that the added complexity of these "projected volumes" is not worth separating the configuration files into separate ConfigMaps/Secrets.
Anyway, thank you for approving the PR and I can now continue working on the deployment script.